lib/perl/Genome/Qc/Tool/Picard.pm

```diff
@@ -20,9 +20,9 @@ sub get_metrics {
 
     my $file = $self->qc_metrics_file;
     my $gmt_class = $self->gmt_class;
-    my %metrics = $self->metrics;
     my $metric_results = $gmt_class->parse_file_into_metrics_hashref($file);
-    return $self->_get_metrics(\%metrics, $metric_results);
+    my $metric_header_as_key = $gmt_class->can('_metric_header_as_key') ? $gmt_class->_metric_header_as_key : undef;
+    return $self->_flatten_metrics_hash($metric_results, metric_header_as_key => $metric_header_as_key);
```

Comment:
The metric_header_as_key supports 'Pair', 'First of Pair' and 'Second of Pair', but as it exists will not support different levels of accumulation, eg. multiple read groups, samples or libraries.  As-is the parsing logic does not throw an error, but only reports the QC metrics for the final read group.  For testing, an example output file can be found here:
/gscuser/jwalker/git/picard_collect_alignment_summary

In addition `CollectAlignmentSummaryMetrics.t` has only the read group level metrics reported in the `%expected_metrics` hash.  There should be at least a sample and read group level set of metrics.  The test BAM file only contains a single read group.  Using a BAM with multiple read groups might help exemplify the issue faced with our production data.